### PR TITLE
Add `scoring-matrices` Python package

### PR DESCRIPTION
**Pull request by althonos**
The `scoring-matrices` package is a Python package for handling and distributing scoring matrices for biological sequence alignment. Ultimately, it will be used to handle scoring matrices in `pyopal`, `pytrimal`, `pyfamsa`, `pytantan`, etc.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by althonos** [comment]
@BiocondaBot please add label

**Comment by bgruening** [comment]
@BiocondaBot please fetch artifacts

**Comment by bgruening** [comment]
Can you please test the artifacts especially the new ARM builds? Thanks

**Comment by martin-g** [comment]
Testing on Linux ARM64:
```
mgrigorov in 🌐 euler-arm-22 in /tmp/scoring-matrices via 🅒 test123 took 2s 
❯ tar xvf scoring-matrices-0.2.0-py310h0e3f8be_0.tar.bz2                                                                                                                                                   (test123) 
info/run_exports.json
info/test/run_test.py
info/test/run_test.sh
info/hash_input.json
info/index.json
info/licenses/COPYING
info/recipe/meta.yaml.template
info/files
info/recipe/meta.yaml
info/recipe/conda_build_config.yaml
info/paths.json
info/about.json
info/git
lib/python3.10/site-packages/scoring_matrices/matrices.h
lib/python3.10/site-packages/scoring_matrices-0.2.0.dist-info/direct_url.json
lib/python3.10/site-packages/scoring_matrices/lib.cpython-310-aarch64-linux-gnu.so
lib/python3.10/site-packages/scoring_matrices/lib.pyi
lib/python3.10/site-packages/scoring_matrices-0.2.0.dist-info/top_level.txt
lib/python3.10/site-packages/scoring_matrices-0.2.0.dist-info/INSTALLER
lib/python3.10/site-packages/scoring_matrices-0.2.0.dist-info/WHEEL
lib/python3.10/site-packages/scoring_matrices-0.2.0.dist-info/COPYING
lib/python3.10/site-packages/scoring_matrices-0.2.0.dist-info/RECORD
lib/python3.10/site-packages/scoring_matrices-0.2.0.dist-info/METADATA
lib/python3.10/site-packages/scoring_matrices-0.2.0.dist-info/REQUESTED
lib/python3.10/site-packages/scoring_matrices/py.typed
lib/python3.10/site-packages/scoring_matrices/__init__.py
lib/python3.10/site-packages/scoring_matrices/tests/__init__.py
lib/python3.10/site-packages/scoring_matrices/tests/test_doctest.py
lib/python3.10/site-packages/scoring_matrices/tests/test_scoring_matrix.py
lib/python3.10/site-packages/scoring_matrices/__init__.pxd
lib/python3.10/site-packages/scoring_matrices/matrices.pxd
lib/python3.10/site-packages/scoring_matrices/lib.pxd
lib/python3.10/site-packages/scoring_matrices/__pycache__/__init__.cpython-310.pyc
lib/python3.10/site-packages/scoring_matrices/tests/__pycache__/__init__.cpython-310.pyc
lib/python3.10/site-packages/scoring_matrices/tests/__pycache__/test_doctest.cpython-310.pyc
lib/python3.10/site-packages/scoring_matrices/tests/__pycache__/test_scoring_matrix.cpython-310.pyc
lib/python3.10/site-packages/scoring_matrices/lib.pyx

mgrigorov in 🌐 euler-arm-22 in /tmp/scoring-matrices via 🅒 test123 
❯ file lib/python3.10/site-packages/scoring_matrices/lib.cpython-310-aarch64-linux-gnu.so                                                                                                                  (test123) 
lib/python3.10/site-packages/scoring_matrices/lib.cpython-310-aarch64-linux-gnu.so: ELF 64-bit LSB shared object, ARM aarch64, version 1 (SYSV), dynamically linked, not stripped

mgrigorov in 🌐 euler-arm-22 in /tmp/scoring-matrices via 🅒 test123 
❯ conda install ./scoring-matrices-0.2.0-py310h0e3f8be_0.tar.bz2                                                                                                                                           (test123) 

Downloading and Extracting Packages:
                                                                                                                                                                                                                     

Downloading and Extracting Packages:

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

mgrigorov in 🌐 euler-arm-22 in /tmp/scoring-matrices via 🅒 test123 took 5s 
❯ python -m unittest -vv scoring_matrices.tests                                                                                                                                                            (test123) 
__init__ (scoring_matrices.lib.ScoringMatrix)
Doctest: scoring_matrices.lib.ScoringMatrix.__init__ ... ok
from_diagonal (scoring_matrices.lib.ScoringMatrix)
Doctest: scoring_matrices.lib.ScoringMatrix.from_diagonal ... ok
from_name (scoring_matrices.lib.ScoringMatrix)
Doctest: scoring_matrices.lib.ScoringMatrix.from_name ... ok
is_integer (scoring_matrices.lib.ScoringMatrix)
Doctest: scoring_matrices.lib.ScoringMatrix.is_integer ... ok
max (scoring_matrices.lib.ScoringMatrix)
Doctest: scoring_matrices.lib.ScoringMatrix.max ... ok
min (scoring_matrices.lib.ScoringMatrix)
Doctest: scoring_matrices.lib.ScoringMatrix.min ... ok
shuffle (scoring_matrices.lib.ScoringMatrix)
Doctest: scoring_matrices.lib.ScoringMatrix.shuffle ... ok
__init__ (line 175) (scoring_matrices.lib.__test__.ScoringMatrix)
Doctest: scoring_matrices.lib.__test__.ScoringMatrix.__init__ (line 175) ... ok
from_diagonal (line 104) (scoring_matrices.lib.__test__.ScoringMatrix)
Doctest: scoring_matrices.lib.__test__.ScoringMatrix.from_diagonal (line 104) ... ok
from_name (line 34) (scoring_matrices.lib.__test__.ScoringMatrix)
Doctest: scoring_matrices.lib.__test__.ScoringMatrix.from_name (line 34) ... ok
is_integer (line 390) (scoring_matrices.lib.__test__.ScoringMatrix)
Doctest: scoring_matrices.lib.__test__.ScoringMatrix.is_integer (line 390) ... ok
max (line 460) (scoring_matrices.lib.__test__.ScoringMatrix)
Doctest: scoring_matrices.lib.__test__.ScoringMatrix.max (line 460) ... ok
min (line 440) (scoring_matrices.lib.__test__.ScoringMatrix)
Doctest: scoring_matrices.lib.__test__.ScoringMatrix.min (line 440) ... ok
shuffle (line 480) (scoring_matrices.lib.__test__.ScoringMatrix)
Doctest: scoring_matrices.lib.__test__.ScoringMatrix.shuffle (line 480) ... ok
test_eq (scoring_matrices.tests.test_scoring_matrix.TestScoringMatrix) ... ok
test_from_diagonal (scoring_matrices.tests.test_scoring_matrix.TestScoringMatrix) ... ok
test_from_diagonal_invalid_length (scoring_matrices.tests.test_scoring_matrix.TestScoringMatrix) ... ok
test_from_name_blosum50 (scoring_matrices.tests.test_scoring_matrix.TestScoringMatrix) ... ok
test_from_name_blosum62 (scoring_matrices.tests.test_scoring_matrix.TestScoringMatrix) ... ok
test_from_name_invalid_name (scoring_matrices.tests.test_scoring_matrix.TestScoringMatrix) ... ok
test_from_str (scoring_matrices.tests.test_scoring_matrix.TestScoringMatrix) ... ok
test_init_empty (scoring_matrices.tests.test_scoring_matrix.TestScoringMatrix) ... ok
test_init_invalid_length (scoring_matrices.tests.test_scoring_matrix.TestScoringMatrix) ... ok
test_list (scoring_matrices.tests.test_scoring_matrix.TestScoringMatrix) ... ok
test_memoryview (scoring_matrices.tests.test_scoring_matrix.TestScoringMatrix) ... ok
test_pickle (scoring_matrices.tests.test_scoring_matrix.TestScoringMatrix) ... ok
test_shuffle_empty (scoring_matrices.tests.test_scoring_matrix.TestScoringMatrix) ... ok
test_shuffle_invalid_alphabet (scoring_matrices.tests.test_scoring_matrix.TestScoringMatrix) ... ok

----------------------------------------------------------------------
Ran 28 tests in 0.026s

OK
```

LGTM!

File: recipes/scoring-matrices/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: cb41deca61bc113b6153c8bc18a1c0aa16729c881e0a8d3a464802e1feb897a4
+
+build:
+  number: 0
+  skip: True  # [py < 36]
+  script: "{{ PYTHON }} -m pip install . --no-deps --no-build-isolation --no-cache-dir --use-pep517 -vvv"
+  run_exports:
+    - {{ pin_subpackage("scoring-matrices", max_pin="x.x") }}
```
Comment (martin-g) [inline recipes/scoring-matrices/meta.yaml:18]:
```suggestion
    - {{ pin_subpackage(name, max_pin="x.x") }}
```

**Review by martin-g — APPROVED** [review]

**Comment by althonos** [comment]
Thank you both!

**Pull request merged**